block/block.go

```diff
@@ -130,8 +130,12 @@ func (m *Manager) attemptApplyCachedBlocks() error {
 		if !blockExists {
 			break
 		}
+		if err := m.validateBlock(cachedBlock.Block, cachedBlock.Commit); err != nil {
```

Comment:
thinking we should delete it and not keep it in cache.

Reply:
block deleted from cache if invalid.